sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -7866,6 +7874,14 @@ object functions {
    */
   def nvl2(col1: Column, col2: Column, col3: Column): Column = Column.fn("nvl2", col1, col2, col3)
 
+  /**
+   * Returns zero if `col` is null, or `col` otherwise.
+   *
+   * @group conditional_funcs
+   * @since 4.0.0
+   */
+  def zeroifnull(col: Column): Column = Column.fn("zeroifnull", col)
```

Comment:
Would need to implement this for Spark Connect in Scala too (e.g., `connector/connect/client/.../functions.scala`)